README.md

```diff
@@ -207,23 +220,293 @@ More information can be found [here](http://www.diveintopython3.net/strings.html
+                modifyClasses = selectClasses)
+```
+```r
+generateRWrappers(pyPkg = "synapseclient",
+                  module = "synapseclient.entity",
+                  setGenericCallback = callback,
+                  modifyClasses = selectClasses)
+```
+
+### Expose functions within a singleton object
+
+In rare cases, we want to expose a set Python functions which are an object's methods, but without exposing the object itself. For example, in the `synapser` package we wish to make available the methods in `synapseclient.client.Synapse` without requiring the R user to instantiate the `Synapse` object. The following will create a singleton object at package load time and expose the object's methods to be called directly.
+
+In a Python session, we would do the following:
+```
+  syn = synapseclient.Synapse()
+  syn.login()
+  syn.get()
+```
+
+And in R, users would not need to know about the `Synapse` object:
+```
+  synLogin()
+  synGet()
+```
+
+Note that:
+* All function calls in R access a common underlying Python object `Synapse`
+* In the following example, the function names in R will be prepended with a `syn` prefix.
+
+For generated .Rd files, we used `functionPrefix` parameter as follows:
+```r
+generateRdFiles(srcRootDir,
+                pyPkg = "synapseclient",
+                module = "synapseclient.client.Synapse",
+                functionPrefix = "syn")
+```
+
+To generate the R wrappers, we need to instantiate the Python object in `zzz.R` file. Then we pass the object name to `generateRWrapper` via `pySingletonName` parameter.
+
+```r
+.onLoad <- function(libname, pkgname) {
+  pyImport("synapseclient")
+  pyExec("syn = synapseclient.Synapse()")
+
+  generateRWrappers(pyPkg = "synapseclient",
+                    module = "synapseclient.client.Synapse",
+                    setGenericCallback = callback,
+                    pySingletonName = "syn",
+                    functionPrefix = "syn")
+}
+```
+
+### Override the returned object in R
+
+You may wish to intercept and modify the values returned by the auto-generated R functions. In the the following example we wish to intercept certain returned R6 objects and modify their class names.
+```r
+objectDefinitionHelper <- function(object) {
+  # change returned object name from "GeneratorWrapper.<some function name>" to "GeneratorWrapper"
+  if (grepl("^GeneratorWrapper", class(object)[1])) {
+    class(object)[1] <- "GeneratorWrapper"
+  }
+  object
+}
+
+generateRWrappers(pyPkg = "synapseclient",
+                  module = "synapseclient.client.Synapse",
+                  setGenericCallback = callback,
+                  functionPrefix = "syn",
+                  pyObjectName = "syn",
+                  transformReturnObject = objectDefinitionHelper)
+```
+
+### Notes on `generateRdFiles`
+
+This helper function can be used to convert Python [Sphinx](http://www.sphinx-doc.org/en/master/) docs into Rd files. It only supports a subset of Sphinx tags, including:
```

Comment:
This function converts Python docstrings into Rd files, recognizing a subset of [Sphinx](http://www.sphinx-doc.org/en/master/) tags: